src/activity_actor.cpp

```diff
@@ -2351,6 +2351,50 @@ void reload_activity_actor::finish( player_activity &act, Character &who )
     } else {
         add_msg( m_neutral, _( "You reload the %1$s with %2$s." ), reloadable_name, ammo_name );
     }
+    item_location loc = reload_targets[0];
+    tripoint point = loc.position();
+    // Reload may have caused the item to increase in size more than the pocket can contain.
+    // We want to avoid this because items will be deleted on a save/load.
+    if( loc.where() == item_location::type::container &&
+        loc.held_by( who ) ) {
+        item_pocket pocket = reloadable.get_containing_pocket( who );
+        if( pocket.remaining_volume() < units::volume() ||
+            pocket.remaining_weight() < units::mass() ) {
+            if( !who.is_armed() ) {
+                add_msg( m_neutral, _( "The %s no longer fits in your inventory so you wield it instead." ),
+                         reloadable_name );
+                who.wield( reloadable );
+            } else {
+                item replace = reloadable;
+                who.remove_item( reloadable );
+                add_msg( m_neutral, _( "The %s no longer fits in your inventory so you drop it instead." ),
+                         reloadable_name );
+                get_map().add_item_or_charges( point, replace );
+            }
+        }
+        // Check if in a vehicle.
+    } else if( loc.where() == item_location::type::vehicle ) {
+        const cata::optional<vpart_position> vp = get_map().veh_at( point );
+        const cata::optional<vpart_reference> vp_cargo = vp->part_with_feature( "CARGO", true );
+        units::volume capacity = vp->vehicle().free_volume( vp_cargo->part_index() );
+        if( capacity < units::volume() ) {
+            item replace = *loc.get_item();
+            loc.remove_item();
+            add_msg( m_neutral, _( "The %s no longer fits in the storage compartment so you drop it instead." ),
+                     replace.tname() );
```

Comment:
Accessing item after it was removed using `loc.remove_item();` will cause a crash.

Reply:
The intention is for replace to be a copy. I'll take another look at it.

Reply:
My bad, you're right. I mistook `item` for `item &`.